src/Microsoft.Identity.Web/ClientInfo.cs

```diff
@@ -29,7 +29,7 @@ public static ClientInfo CreateFromJson(string clientInfo)
             {
                 return DeserializeFromJson<ClientInfo>(Base64UrlHelpers.DecodeToBytes(clientInfo));
             }
-            catch (Exception exc)
+            catch (Exception)
             {
                 throw new ArgumentException($"Failed to parse the returned client info. ", nameof(clientInfo));
```

Comment:
I know this isn't your code, but ArgumentException is really not the appropriate exception here. `ArgumentException`  tells the developer that they did smth wrong :). Parsing client_info, a piece of JSON returned by EVO, has nothing to do with this. 

Personally I would just let the JSON exception flow out of the deserialize (i.e. remove the try / catch). 

In general, it is a good practice to have your own dedicated exceptions for this project, similar to how MSAL has. I suggest that you review the exception strategy before releasing, as changing exceptions is breaking. CC @jmprieur

Reply:
we will discuss. thanks for putting this on our radar.

---
In reply to: [386016125](https://github.com/AzureAD/microsoft-identity-web/pull/39#discussion_r386016125) [](ancestors = 386016125)